src/components/Accounts/Actions/ActionModal.js

```diff
@@ -0,0 +1,513 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  Field,
+  reduxForm,
+  change,
+} from 'redux-form';
+import SafeHTMLMessage from '@folio/react-intl-safe-html';
+import {
+  Row,
+  Col,
+  Button,
+  TextArea,
+  Modal,
+  TextField,
+  Checkbox,
+  Select,
+} from '@folio/stripes/components';
+
+import { calculateSelectedAmount } from '../accountFunctions';
+import css from './PayWaive.css';
+
+const validate = (values, props) => {
+  const {
+    accounts,
+    action,
+    commentRequired
+  } = props;
+
+  const selected = calculateSelectedAmount(accounts);
+  const errors = {};
+
+  if (!values.amount) {
+    errors.amount = <FormattedMessage id="ui-users.accounts.error.field" />;
+  }
+  if (values.amount <= 0) {
+    errors.amount = <FormattedMessage id={`ui-users.accounts.${action}.error.amount`} />;
+  }
+  if (!values.method) {
+    errors.method = <FormattedMessage id="ui-users.accounts.error.select" />;
+  }
+  if (commentRequired && !values.comment) {
+    errors.comment = <FormattedMessage id="ui-users.accounts.error.comment" />;
+  }
+  if (parseFloat(values.amount) > parseFloat(selected)) {
+    errors.amount = <FormattedMessage id={`ui-users.accounts.${action}.error.exceeds`} />;
+  }
+
+  return errors;
+};
+
+const onChange = (values, dispatch, props, prevValues) => {
+  if (values.ownerId !== prevValues.ownerId) {
+    dispatch(change('payment-many-modal', 'method', null));
+  }
+};
+
+const asyncValidate = (values, dispatch, props, blurredField) => {
+  if (blurredField === 'amount') {
+    const amount = parseFloat(values.amount || 0).toFixed(2);
+    dispatch(change(props.form, 'amount', amount));
+  }
+  return new Promise(resolve => resolve());
+};
+
+class ActionModal extends React.Component {
+  static propTypes = {
+    onClose: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    open: PropTypes.bool,
+    accounts: PropTypes.arrayOf(PropTypes.object),
+    data: PropTypes.arrayOf(PropTypes.object),
+    balance: PropTypes.number,
+    initialize: PropTypes.func,
+    submitting: PropTypes.bool,
+    invalid: PropTypes.bool,
+    pristine: PropTypes.bool,
+    reset: PropTypes.func,
+    commentRequired: PropTypes.bool,
+    owners: PropTypes.arrayOf(PropTypes.object),
+    feefines: PropTypes.arrayOf(PropTypes.object),
+    label: PropTypes.string,
+    form: PropTypes.string,
+    action: PropTypes.string,
+    intl: PropTypes.object,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      amount: '0.00',
+      showNotify: false,
+      notify: false,
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
This looks like an anti pattern. Please avoid using `initialize` and also avoid using local state. The state should be already maintained by the `redux-form`. Instead please use `initialValues` when initializing the form from the higher level component. There is also a setting in `redux-form` called `enableReinitialize` which should help here. Also this PR may help more with the setup: https://github.com/folio-org/ui-users/commit/9444e9fede2b8e2c45857cc424122f7d67842a56

In general please avoid using `shouldComponentUpdate` and `componentDidUpdate`.

Reply:
We made this correction in the code.

Reply:
We made the suggested corrections, @mkuklis checked the operation of patron blocks after the changes you added, we realized that it is not working normally, when trying to deactivate the checkboxes, they are not deactivated.
Could you please check this part of the code?

Reply:
Thank you for catching this @DGB-UNAM-FOLIO. This is my fault. I will try to address it today.

Reply:
Ok @DGB-UNAM-FOLIO this has been fixed here: https://github.com/folio-org/ui-users/pull/927/files

Reply:
Do not worry @mkuklis! Thank you for reviewing it.